avocado/plugins/diff.py

```diff
@@ -0,0 +1,387 @@
+            tmp_file = tempfile.NamedTemporaryFile(prefix=prefix,
+                                                   suffix='.html',
+                                                   delete=False)
+
+            setattr(args, 'html_output', tmp_file.name)
+
+        if getattr(args, 'html_output', None) is not None:
+            try:
+                html_diff = difflib.HtmlDiff()
+                html_diff._legend = """
+                    <table class="diff" summary="Legends">
+                    <tr> <td> <table border="" summary="Colors">
+                    <tr><th> Colors </th> </tr>
+                    <tr><td class="diff_add">&nbsp;Added&nbsp;</td></tr>
+                    <tr><td class="diff_chg">Changed</td> </tr>
+                    <tr><td class="diff_sub">Deleted</td> </tr>
+                    </table></td>
+                    <td> <table border="" summary="Links">
+                    <tr><th colspan="2"> Links </th> </tr>
+                    <tr><td>(f)irst change</td> </tr>
+                    <tr><td>(n)ext change</td> </tr>
+                    <tr><td>(t)op</td> </tr>
+                    </table></td> </tr>
+                    </table>"""
+
+                job_diff_html = html_diff.make_file(job1_results,
+                                                    job2_results,
+                                                    fromdesc=job1_id,
+                                                    todesc=job2_id)
+                with open(args.html_output, 'w') as f:
+                    f.writelines(job_diff_html)
+
+                sys.stdout.write('\n--\n%s\n' % args.html_output)
+
+            except Exception as e:
+                LOG.error(e.message)
+                sys.exit(exit_codes.AVOCADO_FAIL)
+
+        if getattr(args, 'open_browser', False):
+            setsid = getattr(os, 'setsid', None)
+            if not setsid:
+                setsid = getattr(os, 'setpgrp', None)
+            inout = file(os.devnull, "r+")
+            cmd = ['xdg-open', args.html_output]
+            subprocess.Popen(cmd, close_fds=True, stdin=inout, stdout=inout,
+                             stderr=inout, preexec_fn=setsid)
+
+    def _validate_filters(self, string):
+        diff_filters = string.split(',')
+        valid_filters = ['all',
+                         'cmdline',
+                         '-cmdline',
+                         'time',
+                         '-time',
+                         'variants',
+                         '-variants',
+                         'results',
+                         '-results',
+                         'config',
+                         '-config',
+                         'sysinfo',
+                         '-sysinfo']
+        add_all = False
+        for diff_filter in diff_filters:
+            if diff_filter in valid_filters:
+                if diff_filter.startswith('-') and 'all' not in diff_filters:
+                    add_all = True
```

Comment:
Although how about using `set` instead of `list` and split the configs to in/out sections?

```python
input_filter = set(value.split(','))
include_options = ["cmdline", "time", ...]
exclude_options = ["nocmdline", "notime", ...]
invalid = input_filter.difference(include_options + exclude_options + ["all"]):
if invalid:
    raise ...
if input_filter.intersection(exclude_options):    # exclude has priority and assumes everything on except the filter
    output_filter = [_ for _ in include_options if ("no" + _) not in input_filter]
elif "all" in input_filter:
    output_filter = include_options
else:
    output_filter = input_filter
```